version: "3.9"
services:
  database:
    image: mysql:latest
    restart: on-failure
    environment:
      - "MYSQL_USER=admin"
      - "MYSQL_PASSWORD=b446b342-608c-11ee-8c99-0242ac120002"
      - "MYSQL_DATABASE=contrnastup"
      - "MYSQL_ROOT_PASSWORD=9631a2e4-608d-11ee-8c99-0242ac120002"
    volumes:
      - "./db/mysql:/var/lib/mysql"
      - "./db-init.sql:/docker-entrypoint-initdb.d/create_tables.sql"
    ports:
      - "${DB_EXTERNAL_PORT:-3306}:3306"
    networks:
      - general

  api:
    build: ./server
    restart: always
    volumes:
      - "./server:/var/www/html"
    ports:
      - "${API_EXTERNAL_PORT:-8080}:8080"
    networks:
      - general

  client:
    build: ./client
    volumes:
      - "./client:/code"

  phpmyadmin:
    image: phpmyadmin
    restart: always
    ports:
      - "${PMA_EXTERNAL_PORT:-8081}:80"
    environment:
      - "MYSQL_ROOT_PASSWORD=9631a2e4-608d-11ee-8c99-0242ac120002"
      - "PMA_HOST=database"
      - "PMA_USER=root"
      - "PMA_PASSWORD=9631a2e4-608d-11ee-8c99-0242ac120002"
    networks:
      - general

  nginx:
    image: nginx:latest
    restart: always
    volumes:
      - "./client/build/:/etc/nginx/src/"
      - "./nginx/nginx.conf:/etc/nginx/nginx.conf"
      - "./nginx/mime.type:/etc/nginx/mime.type"
    ports:
      - "${NGINX_EXTERNAL_PORT:-80}:80"
    networks:
      - general

networks:
  general:
    driver: bridge